frontend/src/app/models/found-document.ts

```diff
@@ -74,24 +82,28 @@ export class FoundDocument {
     }
 
     addTag(tag: Tag): void {
-        const newTags = this.tags$.value.concat([tag]);
-        this.setTags(newTags);
+        this.tags$.pipe(
+            take(1),
+            map(tags => tags.concat([tag])),
+            mergeMap(tags => this.setTags(tags)),
+            tap(() => this.tagsChanged$.next()),
+        ).subscribe();
     }
 
     removeTag(tag: Tag): void {
-        const newTags = _.without(this.tags$.value, tag);
-        this.setTags(newTags);
+        this.tags$.pipe(
+            take(1),
+            map(tags => _.without(tags, tag)),
+            mergeMap(tags => this.setTags(tags)),
+            tap(() => this.tagsChanged$.next()),
```

Comment:
As a personal preference I would put this in the `subscribe` block, avoid another `pipe` part